sqlmesh/core/snapshot/definition.py

```diff
@@ -669,14 +685,15 @@ def missing_intervals(
             else:
                 break
 
+        missing = []
         for i in range(len(dates)):
             if dates[i] >= end_ts:
                 break
             current_ts = dates[i]
             next_ts = (
                 dates[i + 1]
                 if i + 1 < len(dates)
-                else to_timestamp(self.model.cron_next(current_ts))
+                else to_timestamp(self.model.interval_unit().cron_next(current_ts))
```

Comment:
can we store the interval unit above, it's used many times, also this is a tight loop